### PR TITLE
Additional implementations of MulByRef

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by Kixiron** [comment]
The new trait isn't really needed since we already have `MulByRef`/`Mul`/`MulAssign`

**Comment by mbudiu-vmw** [comment]
The types are different

**Comment by Kixiron** [comment]
`MulByRef` has the exact same signature, except more general
```rust
pub trait MulByRef<Rhs = Self> {
    type Output;

    fn mul_by_ref(&self, other: &Rhs) -> Self;
}
```

**Comment by mbudiu-vmw** [comment]
I guess now it does. I will try to do without it. Let's not merge this yet.

**Comment by mbudiu-vmw** [comment]
Can I please get a review for this?
These new implementations are needed by the SQL compiler.

File: src/algebra/mod.rs [hunk 1]
```diff
@@ -270,6 +270,43 @@ where
     }
 }
 
+/// Trait that allows values to be multiplied by weights.  Usually W
+/// will be a ZRingValue.  This is used for example by SQL's SUM
+/// aggregation, whose general implementation on Z-sets needs to
+/// multiply each value with its weight before summing.
+pub trait MulByWeight<W> {
```
Comment (ryzhyk) [inline src/algebra/mod.rs]: I thought we decided not to introduce this trait.

**Pull request merged**